central/sensor/service/connection/connection_test.go

```diff
@@ -57,6 +87,63 @@ func (c *mockServer) Recv() (*central.MsgFromSensor, error) {
 	return nil, nil
 }
 
+func (s *testSuite) TestSendsScanConfigurationMsgOnRun() {
+
+	ctx := sac.WithAllAccess(context.Background())
+
+	ctrl := gomock.NewController(s.T())
+	mgrMock := clusterMgrMock.NewMockClusterManager(ctrl)
+	pipeline := pipelineMock.NewMockClusterPipeline(ctrl)
+	deduper := mocks.NewMockDeduper(ctrl)
+	stopSig := concurrency.NewErrorSignal()
+
+	hello := &central.SensorHello{
+		SensorVersion: "1.0",
+	}
+
+	eventHandler := newSensorEventHandler(&storage.Cluster{}, "", pipeline, nil, &stopSig, deduper, nil)
+
+	sensorMockConn := &sensorConnection{
+		//clusterID:          fixtureconsts.Cluster1,
+		clusterMgr:         mgrMock,
+		sensorEventHandler: eventHandler,
+		sensorHello:        hello,
+		hashDeduper:        deduper,
+		scanSettingDS:      s.scanConfigDS,
+		sendC:              make(chan *central.MsgToSensor),
+	}
+
+	caps := set.NewSet[centralsensor.SensorCapability]([]centralsensor.SensorCapability{centralsensor.SendDeduperStateOnReconnect}...)
+
+	mgrMock.EXPECT().GetCluster(ctx, gomock.Any()).Return(&storage.Cluster{}, true, nil).AnyTimes()
+	deduper.EXPECT().GetSuccessfulHashes().Return(map[string]uint64{"deployment:1": 0}).Times(1)
+	s.scanConfigDS.EXPECT().GetScanConfigurations(ctx, gomock.Any()).Return(scanConfigs, nil).Times(1)
+
+	server := &mockServer{
+		sentList: make([]*central.MsgToSensor, 0),
+	}
+
+	err := sensorMockConn.Run(ctx, server, caps)
+	s.NoError(err)
+
+	var complianceRequests []*central.ComplianceRequest
+	for _, msg := range server.sentList {
+		if m := msg.GetComplianceRequest(); m != nil {
+			complianceRequests = append(complianceRequests, m)
+		}
+	}
+
+	s.Equal(1, len(complianceRequests))
+	ScanConfigs := complianceRequests[0].GetSyncScanConfigs().GetScanConfigs()
+	s.Equal(1, len(ScanConfigs))
+	UpdateScan := ScanConfigs[0].GetUpdateScan()
+	s.Equal("TestConfigName", UpdateScan.GetScanSettings().GetScanName())
+	Profiles := UpdateScan.GetScanSettings().GetProfiles()
+	s.Equal(1, len(Profiles))
+	s.Equal("TestProfileName", Profiles[0])
+	s.Equal("2 1 * * *", UpdateScan.GetCron())
```

Comment:
Use `Require` before accessing the slice in a fixed position. Otherwise some of these might panic.

Also, I think it would be better to iterate through the scan configs and assert on all of them. I know that you have it fixed to just one but this test would fail if in the future someone changes the `scanConfig` variable.

Reply:
Sure, I've refactored it to go over all configs and added a require there 👍